user-guide/Cloud_Platform/Catalog/Tutorials/Tutorial_Register_Catalog_Version_GitHub_Actions.md

```diff
@@ -0,0 +1,156 @@
+             $uri = "https://api.dataminer.services/api/key-catalog/v1-0/catalog/${{ env.CATALOG_ID }}/register/version"
+
+             # Define the form data (package file, version, and description)
+             $formData = @{
+               file = Get-Item $file.FullName
+               versionNumber = "${{ env.VERSION }}"
+               versionDescription = "New version registered by GitHub Actions pipeline"
+             }
+
+             # Set up the API request to register the new version in the catalog
+             Invoke-RestMethod -Uri $uri -Method Post -Headers @{ 'Ocp-Apim-Subscription-Key' = "${{ secrets.API_TOKEN }}" } -Form $formData
+   ```
+
+## Step 2: Add GitHub secrets
+
+To securely store sensitive information like the Catalog ID, API token, and version number, you will need to add GitHub secrets:
+
+1. Create an [organization key](xref:Managing_DCP_keys#organization-keys) token to authenticate the register version call from the Catalog API:
+
+   1. In the [Admin app](https://admin.dataminer.services/), under *Organization* in the sidebar on the left, select the *Keys* page.
+
+   1. At the top of the page, click *New Key*.
+
+   1. Configure the key with a label of your choice and the permission *Register catalog items*.
+
+      ![Organization Key](~/user-guide/images/tutorial_catalog_registration_create_org_key.png)
+
+   1. Copy the key so you can use it later.
+
+   > [!IMPORTANT]
+   > You need to have the *Owner* role in order to access/create organization keys. See [Changing the role of a dataminer.services user](xref:Changing_the_role_of_a_DCP_user) for information on how to change a role for a user.
+
+1. In your GitHub repository, go to the *Settings* page.
+
+1. In the sidebar on the left, select *Secrets and variables* > *Actions*, and then click the *New repository secret* button.
+
+   ![New repository secret button](~/user-guide/images/tutorial_catalog_registration_new_secret.png)
+
+1. Specify the name `API_TOKEN`, and add the organization key you copied earlier as the value.
+
+## Step 3: Push the workflow file
+
+Commit and push the addition of the *dataminer-catalog-pipeline.yml* file in the *.github/workflows* folder to your repository. You can use the following commands for this:
+
+```bash
+git add .github/workflows/dataminer-catalog-pipeline.yml
+git commit -m "Add GitHub Actions pipeline for catalog version registration"
+git push origin main
+```
+
+## Step 4: Trigger the workflow
+
+To create a new version, push a change to the *main* branch of the repository (or the branch you have defined in the workflow under the *on* section). The GitHub Actions pipeline will automatically be triggered.
+
+You can also manually trigger the workflow by navigating to the *Actions* tab in your repository, selecting the pipeline, and then clicking *Run workflow*.
+
+> [!NOTE]
+> A version can only be registered once. Registration will fail if you try to register an existing version number of a Catalog item.
```

Comment:
I'm not sure if this note is in the right place. If I understand everything correctly, the pipeline will automatically increment the version number each time the workflow is triggered, which means that at this point in the tutorial, the user could hardly try to register the same version twice. I think it would be better to move this note to the Register_Catalog_item.md page, below the intro under the "## Registering a new version with the API" header. What do you think?

Reply:
i agree. moved it to registering a new version.